crates/oxc_transformer/src/common/arrow_function_converter.rs

```diff
@@ -703,6 +728,25 @@ impl<'a> ArrowFunctionConverter<'a> {
         ctx: &mut TraverseCtx<'a>,
     ) -> Option<Expression<'a>> {
         if self.super_methods.is_none() || !call.callee.is_member_expression() {
+            // `super()`
+            // Store the address in case we need to insert `var _this;` after it.
+            if call.callee.is_super() {
+                let scope_id = ctx.current_scope_id();
+                self.super_call_addresses.entry(scope_id).or_insert_with(|| {
+                    ctx.ancestors()
+                        .find(|ancestor| {
+                            matches!(
+                                ancestor,
+                                // const A = super():
+                                Ancestor::VariableDeclarationDeclarations(_)
+                                // super();
+                                | Ancestor::ExpressionStatementExpression(_)
+                            )
+                        })
+                        .unwrap()
+                        .address()
```

Comment:
Unwrap failed here, we need to find all possible statements, and this may not be a good implementation

Reply:
You can use `Ancestor::is_via_statement` for this.

Note it doesn't tell you if that ancestor *is* a statement, but whether its *child* is a statement.

So if you are visiting a `Function`, it could be either an expression or a statement. `ctx.parent().is_via_statement()` will tell you.

To find the closest statement, loop through ancestors, and when `ancestor.is_via_statement()` returns `true`, then the *previous* ancestor you visited is the statement.

The naming `is_via_*` is weird, and the fact that you have to look at the ancestor above is confusing as hell. So it's a terrible API! But it does do what you need.

Reply:
#8031 renames this method to `Ancestor::is_parent_of_statement`. I think that's clearer.

Reply:
I just found that we have these methods!